### PR TITLE
Add `ConflictsWith` and more sections for Rollout

### LINKED ISSUE #777: [Bug] The manually added detail explanation of rollout trait is overriden
**Describe the bug**

The generated ref doc is great, though it will override the manually added detail explanation of rollout trait, i.e.: https://github.com/oam-dev/kubevela/blob/efb76ee5ee16a09aaba4b4e612b3165436852e0e/docs/en/developers/references/traits/rollout.md#how-rollout-works

We need to improve the doc gen, or move that section to somewhere else (for example, the user doc of rollout).

### PR DESCRIPTION
**Pull request by zzxwill**
Added `ConflictsWith` and `How Rollout Works` section to Rollout Reference Docs.
To fix #777

### CODE REVIEW

File: docs/en/developers/references/traits/rollout.md [hunk 1]
```diff
@@ -26,3 +26,29 @@ Name | Description | Type | Required | Default
...
+![alt](../../../../../docs/resources/traffic-shifting-analysis.png)
+
+In this sample, for every `10s`, `5%` traffic will be shifted to canary from the primary, until the traffic on canary reached `50%`. At the mean time, the instance number of canary will automatically scale to `replicas: 2` per configured in Appfile.
+
+
+Based on analysis result of the KPIs during this traffic shifting, a canary will be promoted or aborted if analysis is failed. If promoting, the primary will be upgraded from v1 to v2, and traffic will be fully shifted back to the primary instances. So as result, canary instances will be deleted after the promotion finished.
+
+![alt](../../../../../docs/resources/promotion.png)
+
+> Note: KubeVela's `Rollout` trait is implemented with [Weaveworks Flagger](https://flagger.app/) operator.
```
Comment (ryanzhang-oss) [inline docs/en/developers/references/traits/rollout.md:54]: We are not implementing the rollout trait based on flagger.

File: hack/references/generate.go [hunk 2]
```diff
@@ -92,7 +95,13 @@ func main() {
 			os.Exit(1)
 		}
 		specification := fmt.Sprintf("\n\n## Specification\n\n%s\n\n%s", specificationIntro, specificationContent)
-		refContent = title + description + specification + refContent
+
+		conflictWithAndMoreSection, err := generateConflictWithAndMore(capName)
```
Comment (ryanzhang-oss) [inline hack/references/generate.go:99]: Do we need to write new code for each new feature in order to generate its corresponding document?

Comment (zzxwill) [reply hack/references/generate.go:99]: No, run `hack/references/generate.go` will automatically generate all reference docs of those definitions by `vela workloads/traits`. As the generation tool doesn't cover `conflictsWith` and `How to use Rollout` section for Rollout Trait, I have to write new code for it.

File: docs/en/developers/references/traits/rollout.md (hunk 1 above)
Comment (zzxwill) [reply docs/en/developers/references/traits/rollout.md:54]: @wonderflow Please help review Ryan' comments. Thanks.

Comment (wonderflow) [reply docs/en/developers/references/traits/rollout.md:54]: Yes, @ryanzhang-oss  is implementing a new rollout #787,  we can hold this part.

Comment (resouer) [reply docs/en/developers/references/traits/rollout.md:54]: This is end user facing documentation regarding to current status, it's exactly Fagger until the new AppDeployment is released. We should not hold this.

**Review by wonderflow — APPROVED** [review]

File: docs/en/developers/references/traits/rollout.md (hunk 1 above)
Comment (wonderflow) [reply docs/en/developers/references/traits/rollout.md:54]: alright, lgtm and approved

**Comment by zzxwill** [comment]
Thanks for the review:)

**Pull request merged**